lib/models/blueprint.js

```diff
@@ -1027,12 +1027,14 @@ let Blueprint = CoreObject.extend({
   addBowerPackagesToProject(packages, installOptions) {
     let task = this.taskFor('bower-install');
     let installText = (packages.length > 1) ? 'install bower packages' : 'install bower package';
-    let packageNames = [];
-    let packageNamesAndVersions = packages.map(pkg => {
+
+    let packageNames = new Array(packages.length);
+    let packageNamesAndVersions = new Array(packages.length);
+    packages.forEach((pkg, i) => {
       pkg.source = pkg.source || pkg.name;
-      packageNames.push(pkg.name);
-      return pkg;
-    }).map(bowEpParser.compose);
+      packageNames[i] = pkg.name;
+      packageNamesAndVersions[i] = bowEpParser.compose(pkg);
+    });
```

Comment:
@bekzod I'm not sure what the goal of this PR is. I'd personally prefer it like this:

```js
packages.forEach(pkg => (pkg.source = pkg.source || pkg.name));

let packageNames = packages.map(pkg => pkg.name);
let packageNamesAndVersions = packages.map(bowEpParser.compose);
```

It might not be as fast, but is a lot clearer IMHO